frontend/packages/core/src/Input/select.tsx

```diff
@@ -218,20 +220,26 @@ export const Select = ({
     }
 
     if (Number.isInteger(option)) {
-      return option < flatOptions.length && option > 0 ? option : 0;
+      return option < flatOptions.length && option > 0 ? [option] : [0];
     }
 
     // we're a string, lets look it up based on the value/label and default to 0 if none
     const index = flatOptions?.findIndex(opt => opt?.value === option || opt?.label === option);
 
-    return index >= 0 ? index : 0;
+    return index >= 0 ? [index] : [0];
   };
 
-  const [selectedIdx, setSelectedIdx] = React.useState(calculateDefaultOption());
+  const [selectedIdxs, setSelectedIdxs] = React.useState(calculateDefaultOption());
+
+  // Returns the list of selected values
+  const selectedValues = () => {
+    return selectedIdxs.map(idx => flatOptions[idx].value || flatOptions[idx].label);
+  };
 
   React.useEffect(() => {
     if (flatOptions.length !== 0) {
-      onChange && onChange(flatOptions[selectedIdx]?.value || flatOptions[selectedIdx].label);
+      const onChangeInput: string[] = selectedValues();
+      onChange && onChange(onChangeInput.join(","));
```

Comment:
Probably dont need this variable

```suggestion
      onChange && onChange((selectedValues() ?? []).join(","));
```